packages/@aws-cdk/aws-dax/lib/parameter-group.ts

```diff
@@ -0,0 +1,103 @@
+import { Construct, IResource, Lazy, Resource } from '@aws-cdk/core';
+import { CfnParameterGroup } from './dax.generated';
+
+/**
+ * A parameter group
+ */
+export interface IParameterGroup extends IResource {
+  /**
+   * The name of this parameter group
+   *
+   * @attribute
+   */
+  readonly parameterGroupName: string;
+}
+
+/**
+ * A new cluster or instance parameter group
+ */
+abstract class ParameterGroupBase extends Resource implements IParameterGroup {
+  /**
+   * Imports a parameter group
+   */
+  public static fromParameterGroupName(scope: Construct, id: string, parameterGroupName: string): IParameterGroup {
+    class Import extends Resource implements IParameterGroup {
+      public readonly parameterGroupName = parameterGroupName;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The name of the parameter group
+   */
+  public abstract readonly parameterGroupName: string;
+
+  /**
+   * Parameters of the parameter group
+   */
+  protected parameters?: { [key: string]: string } = {};
+
+  /**
+   * Add a parameter to this parameter group
+   *
+   * @param key The key of the parameter to be added
+   * @param value The value of the parameter to be added
+   */
+  public addParameter(key: string, value: string) {
+    if (!this.parameters) {
+      this.parameters = {};
+    }
+    this.parameters[key] = value;
+  }
```

Comment:
This API is awesome (it future proofs the solution), however I'd like to have type-safe APIs for the parameters that are known to exist (for example, for the TTLs that accept `Duration` values).